vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,471 @@
+hub and is also one of the most complex aspects of a hub. This configuration
+file is a formal representation of an official collaborative forecasting
+challenge that allows direct and unambiguous evaluation and comparison of model
+outputs. It allows hub administrators to set a clear standard for model
+submissions.
+
+While we provide [a hubTemplate](https://github.com/hubverse-org/hubTemplate/)
+for use to get started with setting up a hub, JSON files are easy for a computer
+to read, but not so easy for a human to read and less so to write. The hard part
+of creating a `tasks.json` file should be what model parameters to use, not
+constantly looking at JSON syntax or hunting down a missing comma.
+To make this process easier, **we provide the `create_*()` family of functions
+for creating `tasks.json` files programmatically**.  These can be useful for
+creating your initial `task.json` file, amending an existing file's
+configurations or even appending new rounds programmatically. By making use of
+this functionality, hubs can even automate the process of updating their files
+through Github Actions.  For example, the [variant nowcast
+hub](https://github.com/reichlab/variant-nowcast-hub/tree/main) uses a [GitHub
+Action
+workflow](https://github.com/reichlab/variant-nowcast-hub/blob/29a9265d0b30f385fc47195d3d7f65ecda7a41d2/.github/workflows/create-modeling-round.yaml#L47)
+that [runs a
+script](https://github.com/reichlab/variant-nowcast-hub/blob/main/src/make_round_config.R)
+to append a new round with updated variants to model on a weekly schedule!
+
+In this vignette, we will use the config creation functions in `hubAdmin` to
+create a `tasks.json` file that represents two rounds of a modeling effort to
+predict influenza hospitalizations and optionally deaths 1 to 4 weeks ahead
+with mean, median, and quantile predictions in the US, and optionally 5 states.
+
+Before getting to the scripting, it's important to get a general look at what
+a `tasks.json` file looks like structurally.
+
+## Structure of a `tasks.json` configuration file
+
+The structure of a `tasks.json` configuration file roughly looks like the
+diagram below:
+
+```
+tasks.json
+├─schema_version: "https://.../v4.0.0/tasks-schema.json"
+├─rounds:
+│ ├─model_tasks:
+│ │   ├─task_ids:
+│ │   │ └─[...]
+│ │   ├─output_type:
+│ │   │ └─[...]
+│ │   └─target_metadata:
+│ │     └─[...]
+│ ├─round_id_from_variable: true
+│ ├─round_id: origin_date
+│ └─submissions_due:
+│   └─[...]
+├─output_type_id_datatype: "auto"
+└─derived_task_ids:
+  └─[...]
+```
+
+The first property in a `tasks.json` file is the `schema_version`, which provides
+a URL to the [hubverse schema](https://github.com/hubverse-org/schemas) version the config file is built against. The schema is used to validate the structure and contents of the config file.
+The next property defined are the **rounds** which set the schedule and expected contents of model submissions
```

Comment:
The next properties given the above suggestion is actually `round_id_from_variable` and `round_id`. This doesn't appear to be discussed until a lot later. Maybe we should add a short reference here too?

Reply:
I can do that. It's a bit unfortunate that `round_id_from_variable` appears before `round_id` because the former implies knowledge of the latter.

Reply:
addressed in 0ebd33d8de5309bc801692bdc3fd1ea11cf2abec